### ISSUE TITLE
don't treat TypedDict as invariant if they have the same signature

### ISSUE THREAD

**Issue by ITProKyle** [microsoft/pyright#2592]
**Is your feature request related to a problem? Please describe.**

pyright 1.1.189 changed the detection of TypedDict invariance. This caused previous uses of identical TypedDict to now result in type errors.

I am a heavy user of `boto3-stubs`/`mypy-boto3` which provides type stubs for `boto3`. `boto3` has complex arguments and return value structures resulting in the type annotations for the entire library being quite large. For this reason the `boto3-stubs` is distributed per AWS service. Some of these services contain identical TypedDict for data structures that are repeated across the services (e.g. `mypy_boto3_ec2.type_defs.TagTypeDef`). Some of these TypedDict occur in lists which results in an inability to use the same list across multiple services without ignoring of casting the type.

Sure, this could be solved in the stub library by consolidating all or most of the repeat TypedDict into a central location. However this would not work if someone wanted to use a TypedDict to construct a dict without creating a runtime dependency on this large stub library. It may not be the best usage pattern but I and a few devs I have worked with do it at times.

```python
# mypy_boto3_ec2.type_defs.TagTypeDef & mypy_boto3_cloudformation.type_defs.TagTypeDef & ...
TagTypeDef = TypedDict(
    "TagTypeDef",
    {
        "Key": str,
        "Value": str,
    },
)
```

```python
from typing import TYPE_CHECKING, List, TypeDict

if TYPE_CHECKING:
    from mypy_boto3_ec2.type_defs import TagTypeDef as Ec2TagTypeDef

class TagTypeDef(TypedDict):
    """AWS resource tags."""

    Key: str
    Value: str


def example() -> List[Ec2TagTypeDef]:
    return [TagTypeDef(Key="foo", Value="bar")]

#  TypeVar "_T@list" is invariant
#       Type "TagTypeDef" cannot be assigned to type "TagTypeDef" (reportGeneralTypeIssues)
```

**Describe the solution you'd like**

I would like the above example to not result in a type error.

**Comment by erictraut** [comment]
I've address the issue, and the fix will be in the next release. Apologies for any inconvenience this caused.

**Comment by erictraut** [comment]
This is included in pyright 1.1.190, which I just published. It will also be included in the next release of pylance.
